prover/src/logup_gkr/mod.rs

```diff
@@ -109,16 +109,21 @@ impl<E: FieldElement> EvaluatedCircuit<E> {
         log_up_randomness: &[E],
     ) -> CircuitLayer<E> {
         let num_fractions = evaluator.get_num_fractions();
+        let periodic_values = evaluator.build_periodic_values::<E::BaseField, E>();
+
         let mut input_layer_wires =
             Vec::with_capacity(main_trace.main_segment().num_rows() * num_fractions);
         let mut main_frame = EvaluationFrame::new(main_trace.main_segment().num_cols());
 
         let mut query = vec![E::BaseField::ZERO; evaluator.get_oracles().len()];
+        let mut periodic_values_row = vec![E::BaseField::ZERO; periodic_values.num_columns()];
         let mut numerators = vec![E::ZERO; num_fractions];
         let mut denominators = vec![E::ZERO; num_fractions];
         for i in 0..main_trace.main_segment().num_rows() {
             let wires_from_trace_row = {
                 main_trace.read_main_frame(i, &mut main_frame);
+                periodic_values.get_periodic_values_at(i, &mut periodic_values_row);
+                evaluator.build_query(&main_frame, &periodic_values_row, &mut query);
 
                 evaluator.build_query(&main_frame, &[], &mut query);
```

Comment:
We should remove that second call to `build_query()`, right?

Reply:
Yep, remnants from the merge conflicts